File: src/main/java/org/polyfrost/hytils/handlers/chat/modules/blockers/DiscordSafetyWarningRemover.java
/*
 * Hytils Reborn - Hypixel focused Quality of Life mod.
 * Copyright (C) 2020, 2021, 2022, 2023  Polyfrost, Sk1er LLC and contributors
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package org.polyfrost.hytils.handlers.chat.modules.blockers;

import org.polyfrost.hytils.config.HytilsConfig;
import org.polyfrost.hytils.handlers.chat.ChatReceiveModule;
import net.minecraft.util.EnumChatFormatting;
import net.minecraftforge.client.event.ClientChatReceivedEvent;
import org.jetbrains.annotations.NotNull;

public class DiscordSafetyWarningRemover implements ChatReceiveModule {
    @Override
    public void onMessageReceived(@NotNull ClientChatReceivedEvent event) {
        String message = EnumChatFormatting.getTextWithoutFormattingCodes(event.message.getUnformattedText());
        if (message.equals(getLanguage().chatDiscordSafetyWarning)) {
            event.setCanceled(true);
        }
    }

    @Override
    public boolean isEnabled() {
        return HytilsConfig.discordSafetyWarning;
    }

    @Override
    public int getPriority() {
        return -1;
    }
}

File: src/main/kotlin/org/polyfrost/hytils/client/handlers/chat/modules/blockers/DiscordSafetyWarningRemover.kt
package org.polyfrost.hytils.client.handlers.chat.modules.blockers

import org.polyfrost.hytils.client.HytilsRebornConfig
import org.polyfrost.hytils.client.events.ChatReceiveEvent
import org.polyfrost.hytils.client.handlers.chat.ChatReceiveModule
import org.polyfrost.hytils.client.data.providers.LanguageData

object DiscordSafetyWarningRemover : ChatReceiveModule {
    override fun onChatReceived(event: ChatReceiveEvent) {
        if (!event.isOverlay && event.unformattedMessage.contains(LanguageData.DISCORD_SAFETY_WARNING)) {
            val message = event.message.copy()
            message.siblings.removeIf { it.string == LanguageData.DISCORD_SAFETY_WARNING }
            // trim any newlines/whitespace-only siblings left behind
            while (message.siblings.isNotEmpty() && message.siblings.last().string.isBlank()) {
                message.siblings.removeLast()
            }
            event.message = message
        }
    }

    override fun isEnabled() = HytilsRebornConfig.removeDiscordSafetyWarning
    override fun getPriority() = -1
}

